### PR TITLE
added rag eval dataset curation notebook using gemini for cluster ana…

### PR DESCRIPTION
**Pull request by kenleejr**
The purpose of this notebook is to go from a raw a set of user-queries from an in-production chatbot or rag system to a refined evaluation dataset. This is achieved through a mix of traditional clustering and using gemini to generate structured outputs from the unstructured chat data.

1. Embed questions
2. Cluster using favorite clustering algorithm
3. Summarize each cluster using gemini, extracting valuable metadata like sentiment, topics, representative questions, etc.

Best,

### CODE REVIEW

File: genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb [hunk 1]
```diff
@@ -0,0 +1,1027 @@
+{
```
Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>could you please format the notebook in the standard template with the various sections as well as some more detailed intro on what this notebook does, why is it important.</p><p>A sample you can utilize is here - <a href="https://github.com/GoogleCloudPlatform/applied-ai-engineering-samples/blob/main/genai-on-vertex-ai/retrieval_augmented_generation/diy_rag_with_vertexai_apis/build_grounded_rag_app_with_vertex.ipynb" target="_blank">https://github.com/GoogleCloudPlatform/applied-ai-engineering-samples/blob/main/genai-on-vertex-ai/retrieval_augmented_generation/diy_rag_with_vertexai_apis/build_grounded_rag_app_with_vertex.ipynb</a></p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>Could we make the variables a bit more clear -</p><p>Project could mean PROJECT_ID/NUMBER. Standard notation usage is preferred.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    csv_filename = "" # @param
</pre><p>is there also a sample CSV that can be provided to run this notebook ? You can use cell 14 in this notebook as reference to craft a short CSV file that users can run through - <a href="https://github.com/GoogleCloudPlatform/applied-ai-engineering-samples/blob/main/genai-on-vertex-ai/vertex_ai_extensions/notebooks/data_science_code_interpreter.ipynb" target="_blank">https://github.com/GoogleCloudPlatform/applied-ai-engineering-samples/blob/main/genai-on-vertex-ai/vertex_ai_extensions/notebooks/data_science_code_interpreter.ipynb</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.    
</pre><p>Lets scrub the unused code if it is not used in the notebook </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #39.        task: str = "RETRIEVAL_DOCUMENT",
</pre><p>Can we add a line explaining what is TASK_TYPE and why RETRIEVAL_DOCUMENT was chosen here ? let's also link it back to our docs - <a href="https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/task-types#retrieve_information_from_texts" target="_blank">https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings/task-types#retrieve_information_from_texts</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>Let's keep some of these texts more generic - eg: a company's internal chat channel</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

**Comment by kenleejr** [comment]
Hi @Abhishekbhagwat I've fixed all the changes you noted, let me know if you would like more.

**Comment by RajeshThallam** [comment]
@kenleejr Can you please add a README with brief description to your folder as well?

File: genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb [hunk 2]
```diff
@@ -0,0 +1,2651 @@
...
```
Comment (RajeshThallam) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>CSV_PATH is not defined</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (RajeshThallam) [inline genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>Can you please add a conclusion about what they did and what can they do next with the eval questions.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/applied-ai-engineering-samples/pull/108/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='47'/>

Comment (kenleejr) [reply genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>Yep added</p>

Comment (kenleejr) [reply genai-on-vertex-ai/vertex_evaluation_services/gemini-curate-evaluation-data/curate_new_evals.ipynb:1]: <p>Yep added.</p>

**Review by Abhishekbhagwat — COMMENTED** [review]
LGTM! Thanks for your contribution :)

**Review by Abhishekbhagwat — APPROVED** [review]
LGTM! Thanks for your contribution :)

**Pull request merged**